src/utils/isSameDay.js

```diff
@@ -2,5 +2,9 @@ import moment from 'moment';
 
 export default function isSameDay(a, b) {
   if (!moment.isMoment(a) || !moment.isMoment(b)) return false;
-  return a.isSame(b, 'day');
+  // Compare least significant, most likely to change units first
+  // Moment's isSame clones moment inputs and is a tad slow
```

Comment:
This seems like something we could PR a fix for into moment itself rather than hacking around it?